lib/OpenQA/Worker/Job.pm

```diff
@@ -568,11 +569,22 @@ sub _format_reason {
         }
     }
     catch {
+        # read autoinst-log.txt to check the reason, see poo#80334
+        my $msg               = '';
+        my $autoinst_log_file = path($self->worker->pool_directory, 'autoinst-log.txt');
+        if (!-e $autoinst_log_file) {
+            log_warning("File $autoinst_log_file does not exist");
+        }
+        else {
+            map_file my $log_content, $autoinst_log_file, '<';
+            $msg = ': No space left on device' if ($log_content =~ /No space left on device/);
+        }
```

Comment:
`map_file` will throw an exception if the file can not be opened. I would rather catch this exception instead of testing this condition before. (There might be other exceptions as well and the file might just vanish between `-e …` and `map_file …`.

```suggestion
        eval {
            map_file my $log_content, path($self->worker->pool_directory, 'autoinst-log.txt'), '<';
            $msg = ': No space left on device' if ($log_content =~ /No space left on device/);
        }
        log_warning $@ if $@;
```